src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData)
+    } else if (action.type === 'keysToRemove') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else {
+      newState = {};
```

Comment:
You cannot reassign a constant variable 'newState'. Use 'let' for 'newState' declaration if you want to reassign it to an empty object in the 'clear' action type.